spark/src/test/java/org/apache/zeppelin/spark/SparkSqlInterpreterTest.java

```diff
@@ -161,6 +163,41 @@ public void test_null_value_in_row() {
     assertEquals("name\tage\ngates\tnull\n", ret.message().get(0).getData());
   }
 
+  private SparkInterpreter getSparkInterpreter() {
+    LazyOpenInterpreter lazy = null;
+    SparkInterpreter spark = null;
+    Interpreter p = sql.getInterpreterInTheSameSessionByClassName(SparkInterpreter.class.getName());
+
+    while (p instanceof WrappedInterpreter) {
+      if (p instanceof LazyOpenInterpreter) {
+        lazy = (LazyOpenInterpreter) p;
+      }
+      p = ((WrappedInterpreter) p).getInnerInterpreter();
+    }
+    spark = (SparkInterpreter) p;
+
+    if (lazy != null) {
+      lazy.open();
+    }
+    return spark;
+  }
+
+  @Test
+  public void testVariableInterpolation() {
+    InterpreterContext newContext = new InterpreterContext("NoteId", "ParaId", null,
+            "testVariableInterpolation", "",
+            null, null, null, null,
+            new LocalResourcePool("testVariableInterpolation"),
+            null, null);
+    SparkZeppelinContext zc = getSparkInterpreter().getZeppelinContext();
+    zc.setInterpreterContext(newContext);
+    zc.put("n", "100");
+    zc.put("table", "name");
+    String commandWithVariables = "select * from {table} where count = {n}";
```

Comment:
@sanjaydasgupta in Jupyter that feature is actually more flexible - that gets eval()ed as Python code. So I can have {var1+100} for example there. I can't find documentation on this feature of Jupyter, but I just ran {non_existent_var} and it does not show errors, it actually returns "{non_existent_var}" as is not substituted at all. I think that's in line with what you were suggesting earlier. See screenshot below. But I agree with @felixcheung  it might be less confusing for users just to show error if a variable is not found.

![image](https://user-images.githubusercontent.com/3013418/28725977-878d2628-737c-11e7-85cf-8efd7336e724.png)